ui/src/components/home/MailingListForm.js

```diff
@@ -0,0 +1,163 @@
+import {
+  Box,
+  Button,
+  Link,
+  Modal,
+  ModalCloseButton,
+  Image,
+  ModalContent,
+  AspectRatio,
+  ModalOverlay,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  VStack,
+  Text,
+  Input,
+  ButtonGroup,
+  useToast,
+} from '@chakra-ui/react';
+import { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { customAxios as axios } from '../helpers/customAxios';
+
+const MailingListForm = ({ props }) => {
+  const user = props;
+  const [isOpen, setIsOpen] = useState(true);
+  const toast = useToast();
+  const onClose = () => {
+    setIsOpen(false);
+  };
+  const inputBox = {
+    color: 'black',
+    background: '#EDF2F7',
+    border: '1px solid #E2E8F0',
+    boxSizing: 'border-box',
+    borderRadius: '12px',
+    flex: 'none',
+    alignSelf: 'stretch',
+    flexGrow: '0',
+    margin: '8px 0px',
+    padding: '15px',
+    fontSize: 'inherit',
+    fontWeight: '500',
+  };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const [error, setError] = useState('');
+
+  const handleSignup = async (data) => {
+    try {
+      const payload = await axios.post('/api/mailingList/create', {
+        email: data.email,
+        category: 'advent',
+      });
+      toast({
```

Comment:
other than showing the toast, should also set the modal to just close so people are not left needing to close the modal